pkg/operator/ceph/pool/radosnamespace/controller.go

```diff
@@ -0,0 +1,312 @@
+		// Build the NamespacedName to fetch the CephBlockPool and make sure it exists, if not we cannot
+		// create the rados namespace
+		cephBlockPool := &cephv1.CephBlockPool{}
+		cephBlockPoolRadosNamespacedName := types.NamespacedName{Name: cephBlockPoolRadosNamespace.Spec.BlockPoolName, Namespace: request.Namespace}
+		err = r.client.Get(r.opManagerContext, cephBlockPoolRadosNamespacedName, cephBlockPool)
+		if err != nil {
+			if kerrors.IsNotFound(err) {
+				return reconcile.Result{}, errors.Wrapf(err, "failed to fetch ceph blockpool %q, cannot create rados namespace %q", cephBlockPoolRadosNamespace.Spec.BlockPoolName, cephBlockPoolRadosNamespace.Name)
+			}
+			// Error reading the object - requeue the request.
+			return reconcile.Result{}, errors.Wrap(err, "failed to get cephBlockPoolRadosNamespace")
+		}
+
+		// If the cephBlockPool is not ready to accept commands, we should wait for it to be ready
+		if cephBlockPool.Status.Phase != cephv1.ConditionReady {
+			// We know the CR is present so it should a matter of second for it to become ready
+			return reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, errors.Wrapf(err, "failed to fetch ceph blockpool %q, cannot create rados namespace %q", cephBlockPoolRadosNamespace.Spec.BlockPoolName, cephBlockPoolRadosNamespace.Name)
+		}
+	}
+	if cephCluster.Spec.External.Enable {
+		logger.Debug("external rados namespace creation is not supported, create it manually, the controller will assume it's there")
+	} else {
+		// Create or Update rados namespace
+		err = r.createOrUpdateRadosNamespace(cephBlockPoolRadosNamespace)
+		if err != nil {
+			if strings.Contains(err.Error(), opcontroller.UninitializedCephConfigError) {
+				logger.Info(opcontroller.OperatorNotInitializedMessage)
+				return opcontroller.WaitForRequeueIfOperatorNotInitialized, nil
+			}
+			r.updateStatus(r.client, request.NamespacedName, cephv1.ConditionFailure)
+			return reconcile.Result{}, errors.Wrapf(err, "failed to create or update ceph pool rados namespace %q", cephBlockPoolRadosNamespace.Name)
+		}
+	}
+	// Update CSI config map
+	// If the mon endpoints change, the mon health check go routine will take care of updating the
+	// config map, so no special care is needed in this controller
+	csiClusterConfigEntry := csi.CsiClusterConfigEntry{
+		Monitors:       csi.MonEndpoints(r.clusterInfo.Monitors),
+		RadosNamespace: cephBlockPoolRadosNamespace.Name,
+	}
+	err = csi.SaveClusterConfig(r.context.Clientset, buildClusterID(cephBlockPoolRadosNamespace), r.clusterInfo, &csiClusterConfigEntry)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "failed to save cluster config")
+	}
+	// Success! Let's update the status
+	if cephCluster.Spec.External.Enable {
+		r.updateStatus(r.client, request.NamespacedName, cephv1.ConditionConnected)
+	} else {
+		r.updateStatus(r.client, request.NamespacedName, cephv1.ConditionReady)
+	}
+
+	// Return and do not requeue
+	logger.Debug("done reconciling")
```

Comment:
The rook operator can operate on multiple resources in multiple namespaces. All info/debug messages must list the namespaced name so context is clear.

```suggestion
	logger.Debug("done reconciling %q", namespacedName)
```